calculatorV2.rb

```diff
@@ -0,0 +1,120 @@
+# Calculator program
+
+# Get operation input from user
+puts "Hey-oh! What kind of calculation would you like to perform?"
+operation = gets.chomp
+
+until operation == "^" || operation == "exp" || operation == "exponent" || operation == "%" || operation == "mod" || operation == "modulo" || operation == "*" || operation == "mult" || operation == "multiply" || operation == "/" || operation == "div" || operation == "divide" || operation == "-" || operation == "sub" || operation == "subtract" || operation == "+" || operation == "add" || operation == "addition"
+	puts "I\'m sorry, I don't understand. Please try again"
+	#kick it back to retrieve new operation and reassign operation
+	operation = gets.chomp
+end 	
+
+
+
+
+# Get first number input from user
+puts "Awesome. What is the first number you would like to perform that on?"
+first_number = gets.chomp
+
+if first_number == "0"
+	first_number_int = first_number.to_i
+
+	elsif first_number.to_i == 0
+		
+		while first_number.to_i == 0
+			first_number_int = first_number.to_i
+
+			if first_number == "0"
+			first_number_int = first_number.to_i
+
+			elsif first_number.to_i == 0
+			puts "Please enter a number."
+			first_number = gets.chomp
+
+			end
+
+		end
+	else
+		first_number_int = first_number.to_i		
+end
+
+# what if they input a float? we don't want to turn it into an integer. We just want to get rid of strings.
+
+# if first_number.is_a? Integer == "false" && first_number.is_a? Float == "false"
+# 	puts "Please enter a number."
+# 	first_number = gets.chomp
+# end		
+# problem here though is if they enter another bad number, it needs to do the loop again to check...
+
```

Comment:
One thing that can be helpful to think about when writing programs is how you can decrease complexity. I like your comments here working through your thoughts about adding floats. One idea that may work for you is just turning all the numbers into floats and dealing with how to present the answer in another step if the answer ends up being an integer and you don't want the trailing ".0".